src/Microsoft.AspNet.JsonPatch/Operations/OperationOfT.cs

```diff
@@ -52,11 +52,5 @@ public void Apply([NotNull] TModel objectToApplyTo, [NotNull] IObjectAdapter ada
             }
```

Comment:
I think both should exists... if the user is developing using the .NET stack with a shared DTO layer between a client (say an MVC app) and the server (RESTful API/Web API), then the preferred way of working with JsonPatchDocument (or with a list of operations to create the doc from) is the generic version. This allows better checking on typo's etc - those can be avoid through creating operations using expressions.  

But there's another use case: one where there's no shared DTO layer, or one where the client might be an MVC app & API is built using another tech: sometimes, users won't create a DTO layer for those cases either => then, the user would typically use the non-generic JsonPatchDocument.  

Both result in the exact same JSON when serialized - but the added convenience (& reliability) of a generic version (as we have already) should be a good reason to keep both.